extensions/arc/src/models/controllerModel.ts

```diff
@@ -60,11 +60,36 @@ export class ControllerModel {
 		};
 	}
 
+	/**
+	 * acquires a login session to this controller. If a login session is in progress against any controller
+	 * then this call waits for the previous login session to finish before acquiring the login session.
+	 * @param promptReconnect - if true then the user is prompted for connection information.
+	 */
+	public async acquireLogin(promptReconnect: boolean = false): Promise<void> {
```

Comment:
What about other callers to azdata calling login? Both from other extensions in ADS and from outside?

We need to decide how much effort we put into preventing this issue - the reality though currently is that we'll never be able to be 100% sure that we aren't changing the login context until we get the context parameter support we've previously discussed. 

What immediate concerns do you have with our current approach such that these changes are necessary now? Currently the code does a login and then immediately runs the command which should avoid most issues unless people are doing multiple actions simultaneously.

Reply:
While the arc controllers' tree is refreshing if you open management the management page of miaa/pg for a controller that has refreshed then results are unpredictable for the arc controller that is yet to refresh.

For other callers to azdata login from other extensions, they would need to take care of this synchronization themselves. In our current codebase, what other similar call are you worried about?

Reply:
Is this something that you've seen while testing? I'm surprised that it'd be that big of issue given the nature of the call sequencing as it is. 

Let's talk this through more before you spend more time on this - it doesn't need to be fixed immediately and I want to make sure we're approaching this correctly.

Reply:
Yes i have observed this in testing and is an easy repro.